### ISSUE TITLE
Crash due to ConcurrentModificationException during Batching chunks

### ISSUE THREAD

**Issue by bcmpinc** [Engine-Room/Flywheel#88]
### Describe the Bug

While playing the game on occasion randomly crashes with a ConcurrentModificationException during Batching chunks. ~~Disabling the flywheel backend seems to prevent these crashes.~~ Edit: Crash still happens. This crash might not be flywheel related.

### Reproduction Steps

This is a relatively rare occurring crash and seems to be caused by a race condition. As such there's not really a reliable way to reproduce this other than waiting. I'm close to a relatively large and dynamic create factory with plenty of chunk updates, which I suspect increases the chance of this crash occurring.

### Expected Result

The game does not crash.

### Crash Report or Log

https://pastebin.com/2CLp2Yk4

### Operating System

Linux (amd64) version 5.11.0-41-generic

### What is your GPU?

Nvidia GeForce RTX 3070

### Mod Version

0.2.5

### Minecraft Version

1.16.5

### Forge Version

36.2.8

### Other Mods

Basically the create above and beyond mod pack version 1.3, with some minor additions, including as FTB Backups and Create Plus.

**Comment by bcmpinc** [comment]
This is most likely a bug in create and not in flywheel. Reducing the speed at which some of my contraption rotate and thus form and are placed back in the world seems to have solved the issue.
